clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1105,6 +1105,19 @@ static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
   return (KHArg != KernelCallerFunc->param_end()) ? *KHArg : nullptr;
 }
 
+static bool isReadOnlyAccessor(const TemplateArgument &AccessModeArg) {
+  const auto *AccessModeArgEnumType =
+      AccessModeArg.getIntegralType()->castAs<EnumType>();
+  EnumDecl *ED = AccessModeArgEnumType->getDecl();
+
+  auto ReadOnly = llvm::find_if(ED->enumerators(), [&](EnumConstantDecl *E) {
+    return E->getName() == "read";
+  });
+
+  return (ReadOnly != ED->enumerator_end()) &&
```

Comment:
```suggestion
  return ReadOnly != ED->enumerator_end() &&
```
Nice, this approach also works well!